quinn/src/connection.rs

```diff
@@ -90,8 +90,10 @@ impl NewConnection {
 /// `Connection` API object to the `Endpoint` task and the related stream-related interfaces.
 /// It also keeps track of outstanding timeouts for the `Connection`.
 ///
-/// If the connection encounters an error condition, this future will yield an error. It
-/// will terminate (yielding `Ready(())`) if the connection was closed without error.
+/// If the connection encounters an error condition, this future will yield an error. It will
+/// terminate (yielding `Ok(())`) if the connection was closed without error. Unlike other
```

Comment:
Not sure `yielding Ok(())` makes sense here, since it really yields `Ok(Async::Ready(()))`?

Reply:
`Future::poll` returns `Ok(Async::Ready(()))`, but if you do `driver.then(|x| ...)` the `x` you get is `Ok(())`. The latter is what the user is typically concerned with, and is what `yield` most precisely refers to.

Reply:
I don't think it's so clear cut (that is, I clearly had/have a different interpretation). Maybe replace `yielding` with `returning` and then use `Ok(Async::Ready(()))` to be fully explicit? I get where you're coming from (and it makes more sense in an async/await world), but it would bother me that the `Ok(())` does not occur in the actual type signature of the old-style `Future::poll()` return type.

Reply:
Note that we're documenting the future itself here, not the `poll` implementation. It'd a weird wart (corrected in futures 0.3, regardless of async/await) that `poll`'s return type has `Result` inlined into it, but it is conventional to refer to the implied `Result` as "the result of the future," as done in e.g. the documentation of [`Future::then`](https://docs.rs/futures/0.1.25/futures/future/trait.Future.html#method.then). Would you be happier with "(resulting in `Ok(())`)"?

Reply:
Fine to leave it as is, then.